src/bp-messages/bp-messages-functions.php

```diff
@@ -116,6 +117,13 @@ function messages_new_message( $args = '' ) {
 
 		$new_reply = true;
 
+		$first_message = BP_Messages_Thread::get_first_message( (int) $thread->thread_id );
+		if ( isset( $first_message->id ) ) {
+			$group = bp_messages_get_meta( $first_message->id, 'group_id', true ); // group id
+			if ( !empty( $group ) ) {
+				$is_group_thread = true;
+			}
+		}
```

Comment:
@buddy4kp First message is already fetched in `$thread` as `$thread->messages[0]`. You can get it without making the query again.